sdk/storage/azure-storage-blobs/src/block_blob_client.cpp

```diff
@@ -196,7 +193,8 @@ namespace Azure { namespace Storage { namespace Blobs {
     };
 
     auto uploadBlockFunc = [&](int64_t offset, int64_t length, int64_t chunkId, int64_t numChunks) {
-      Azure::Core::Http::FileBodyStream contentStream(fileReader.GetHandle(), offset, length);
+      (void)offset;
+      Azure::Core::Http::FileBodyStream contentStream(fileReader.GetHandle(), length);
```

Comment:
@Jinming-Hu - how would we want to update the concurrent upload and download scenario, when the underlying `FileBodyStream` doesn't need offset anymore, and tracks it via the handle automatically?

Assuming we want this API change, I am hoping storage team can react to the changes to `FileBodyStream` and fix up the logic here, since it seems quite complicated for me to tackle without understanding all the edge  cases.

Reply:
>  tracks it via the handle automatically

Handle mustn't contain any information about offset, because we're going to read from the same handle in parallel. This has been discussed during the design of `FileBodyStream` a long time ago.

Reply:
Here're some options I can think of.

1. Disable parallel read, use mutex to make read operations sequential  
2. Duplicate the handle for each thread that's reading the same file. This will cause problem because by default you only have 1024 handles for a process on Linux. You can increase the limit, but you gotta have sudo privileges.
3. undo the changes to `FileBodyStream` in this PR